jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -0,0 +1,339 @@
+        var target = response.target;
+        if (target.status >= 400) {
+          failureCallback(response);
+          return;
+        }
+        successCallback(response);
+      },
+
+      /**
+       Authenticate  by discharging the macaroon and
+       then set the cookie by calling the authCookiePath provided
+
+       @method authenticate
+       @param {Macaroon} The macaroon to be discharged
+       @param {String} The path where to send put request to set the cookie back
+       @param {Function} The request to be sent again in case of
+       successful authentication
+       @param {Function} The callback failure in case of wrong authentication
+       */
+      _authenticate: function (m, authCookiePath,
+                               requestFunction, failureCallback) {
+        try {
+          macaroon.discharge(
+            macaroon.import(m),
+            this._obtainThirdPartyDischarge.bind(this),
+            this._processDischarges.bind(
+              this,
+              authCookiePath,
+              requestFunction,
+              failureCallback
+            ),
+            failureCallback
+          );
+        } catch (ex) {
+          failureCallback(ex.message);
+        }
+      },
+
+      /**
+       Process the discharged macaroon and call the end point to be able to set
+       a cookie for the origin domain only when n auth cooie path i provided,
+       then call the original function.
+
+       @method _processDischarges
+       @param {String} The path where to send put request to set the cookie back
+       @param {Function} The request to be sent again in case of
+       successful authentication
+       @param {Function} The callback failure in case of wrong authentication
+       @param {[Macaroon]} The macaroons being discharged
+       */
+      _processDischarges: function (authCookiePath, requestFunction,
+                                    failureCallback, discharges) {
+        var jsonMacaroon = macaroon.export(discharges);
+        var content = JSON.stringify({'Macaroons': jsonMacaroon});
+        if (authCookiePath === null) {
+          this._setMacaroonsCookie(requestFunction, jsonMacaroon);
+          return;
+        }
+        this.webhandler.sendPutRequest(
+          authCookiePath,
+          null, content, null, null, true, null,
+          this._requestHandler.bind(
+            this,
+            this._setMacaroonsCookie.bind(this, requestFunction, jsonMacaroon),
+            failureCallback
+          )
+        );
+      },
+
+      /**
+       Process succesful discharche by setting MAcaroons Cookie
+       and invoke the original request
+
+       @method _setMacaroonsCookie
+       @param {Function} The path where to send put request
+              to set the cookie back
+       @param {Object} an exported Macaroon
+       */
+      _setMacaroonsCookie: function (originalRequest, jsonMacaroon) {
+        Y.Cookie.set('Macaroons', btoa(JSON.stringify(jsonMacaroon)));
```

Comment:
Is this Macaroons name correct? Doesn't the server look for macaroons with the cookie name "macaroon-*" ?